bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/evaluation/EvaluationInteractor.kt

```diff
@@ -45,39 +47,135 @@ internal class EvaluationInteractor(
+        .commit()
+    }
+
+  init {
+    updateFeatureTag(featureTag)
+  }
+
+  private fun updateFeatureTag(value: String) {
+    // using `this.featureTag` to make it doesn't confuse with the constructor params `featureTag`
+    // https://github.com/bucketeer-io/android-client-sdk/issues/69
+    // 1- Save the featureTag in the SharedPreferences configured in the BKTConfig
+    // 2- Clear the userEvaluationsID in the SharedPreferences if the featureTag changes
+    if (this.featureTag != value) {
+      currentEvaluationsId = ""
+      this.featureTag = value
+    }
+  }
+
+  // https://github.com/bucketeer-io/android-client-sdk/issues/69
+  // userAttributesUpdated: when the user attributes change via the customAttributes interface,
+  // the userAttributesUpdated field must be set to true in the next request.
+  fun setUserAttributesUpdated() {
+    userAttributesUpdated = true
+  }
+
   @Suppress("MoveVariableDeclarationIntoWhen")
   fun fetch(user: User, timeoutMillis: Long?): GetEvaluationsResult {
     val currentEvaluationsId = this.currentEvaluationsId
-
-    val result = apiClient.getEvaluations(user, currentEvaluationsId, timeoutMillis)
+    val condition = UserEvaluationCondition(
+      evaluatedAt = evaluatedAt,
+      userAttributesUpdated = userAttributesUpdated.toString(),
+    )
+    val result = apiClient.getEvaluations(user, currentEvaluationsId, timeoutMillis, condition)
 
     when (result) {
       is GetEvaluationsResult.Success -> {
         val response = result.value
         val newEvaluationsId = response.userEvaluationsId
+
         if (currentEvaluationsId == newEvaluationsId) {
           logd { "Nothing to sync" }
+          // make sure we set `userAttributesUpdated` back to `false` even in case nothing to sync
+          userAttributesUpdated = false
           return result
         }
 
-        val newEvaluations = response.evaluations.evaluations
+        val activeEvaluations: List<Evaluation>
+        var shouldNotifyListener = true
+
+        // https://github.com/bucketeer-io/android-client-sdk/issues/69
+        // forceUpdate: a boolean that tells the SDK to delete all the current data
+        // and save the latest evaluations from the response
+        val forceUpdate = response.evaluations.forceUpdate
+        if (forceUpdate) {
+          // 1- Delete all the evaluations from DB, and save the latest evaluations from the response into the DB
+          activeEvaluations = response.evaluations.evaluations
+        } else {
+          val archivedFeatureIds = response.evaluations.archivedFeatureIds
+          val updatedEvaluations = response.evaluations.evaluations
+          val currentEvaluationsMap = evaluationDao.get(user.id).associateBy { it.id }.toMutableMap()
+          // 1- Check the evaluation list in the response and upsert them in the DB if the list is not empty
+          updatedEvaluations.forEach { evaluation ->
+            currentEvaluationsMap[evaluation.id] = evaluation
```

Comment:
@duyhungtnn cc: @yshrsmz 

The evaluation ID is based on the feature version.
Every time a flag changes, the version number is incremented.

So when the flag is new, it inserts correctly, but when it already exists, it duplicates the flag in the DB.
We will probably need to fix the iOS and JS SDK. We need to check it.

![Screenshot 2023-09-22 at 12 42 36 PM](https://github.com/bucketeer-io/android-client-sdk/assets/2486691/8b13f5b8-5cae-4d91-b540-e8e91f81caf2)

Reply:
https://github.com/bucketeer-io/javascript-client-sdk/blob/main/src/internal/evaluation/EvaluationStorage.ts#L94

JS SDK seems to use featureId, rather than id

Reply:
@yshrsmz  Thanks for confirming

@duyhungtnn 
I confirmed it doesn't duplicate because the evaluation object is a list, so it upserts in the order from old to new.
But we should fix it to use the feature flag id rather than the evaluation id.

### Delete all and Insert

https://github.com/bucketeer-io/android-client-sdk/blob/main/bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/evaluation/db/EvaluationDaoImpl.kt#L99-L102

### Insert 
https://github.com/bucketeer-io/android-client-sdk/blob/main/bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/evaluation/db/EvaluationDaoImpl.kt#L46

Reply:
Hi @cre8ivejp  I fixed this bug in https://github.com/bucketeer-io/android-client-sdk/pull/88/commits/09fab1de9993d098097449aa8fe23b53ecd1c735.
Please help me to take a look.